.github/workflows/deploy_to_do.yml

```
@@ -0,0 +1,155 @@
+            # ==============================
+            TIPTAP_URL=${{ secrets.TIPTAP_URL }}
+            TIPTAP_AUTH_KEY=${{ secrets.TIPTAP_AUTH_KEY }}
+            TIPTAP_JWT_SIGNING_KEY=${{ secrets.TIPTAP_JWT_SIGNING_KEY }}
+
+            # ==============================
+            #   Frontend Configuration
+            # ==============================
+            FRONTEND_IMAGE_NAME=${{ secrets.FRONTEND_IMAGE_NAME }}
+            FRONTEND_CONTAINER_NAME=${{ secrets.FRONTEND_CONTAINER_NAME }}
+            FRONTEND_SERVICE_PORT=${{ secrets.FRONTEND_SERVICE_PORT }}
+
+            # ==============================
+            #   Platform Configuration
+            # ==============================
+            PLATFORM=${{ secrets.PLATFORM }}
+            EOF
+
+            echo "🔧 Setting .env file permissions to match docker-compose.yaml..." # Set permissions
+            chmod --reference=docker-compose.yaml .env # Copy permissions from docker-compose.yaml
+            chown --reference=docker-compose.yaml .env # Copy ownership from docker-compose.yaml
+            
+            echo "📋 Displaying .env file contents (with masked secrets):" # Debug output
+            sed "s/POSTGRES_PASSWORD=.*/POSTGRES_PASSWORD=***/g; s/TIPTAP_AUTH_KEY=.*/TIPTAP_AUTH_KEY=***/g; s/TIPTAP_JWT_SIGNING_KEY=.*/TIPTAP_JWT_SIGNING_KEY=***/g" .env
+            
+            echo "📥 Logging into GHCR..." # Login to container registry
+            echo ${{ secrets.GHCR_PAT }} | docker login ghcr.io -u ${{ secrets.GHCR_USERNAME }} --password-stdin # Login to GHCR
+            
+            echo "📥 Pulling images from .env file..." # Pull images
+            if [ -n "${{ secrets.BACKEND_IMAGE_NAME }}" ]; then
+              echo "Pulling backend image: ${{ secrets.BACKEND_IMAGE_NAME }}"
+              docker pull ${{ secrets.BACKEND_IMAGE_NAME }}
+            fi
+            
+            if [ -n "${{ secrets.FRONTEND_IMAGE_NAME }}" ]; then
+              echo "Pulling frontend image: ${{ secrets.FRONTEND_IMAGE_NAME }}"
+              docker pull ${{ secrets.FRONTEND_IMAGE_NAME }}
+            fi
+            
+            echo "🔍 Verifying docker-compose configuration..." # Verify config
+            docker compose config --quiet # Validate compose file with .env
+            
+            echo "🛑 Stopping existing containers..." # Stop containers
+            docker compose down # Bring down existing services
+
+            echo "🚀 Starting new containers with updated images..." # Start containers
+            docker compose up -d # Deploy in detached mode
+
+            echo "⏳ Waiting for containers to initialize..." # Wait for startup
+            sleep 15 # Brief pause to let containers start
+
+            echo "🩺 Checking container status..." # Check container status
+            docker ps -a # Show all containers
+            
+            echo "🩺 Checking rust-app status..." # Check specific container
+            if docker ps | grep -q rust-app; then
```

```suggestion
            if docker_output=$(docker ps) && echo "$docker_output" | grep -q rust-app && echo "$docker_output" | grep -q nextjs-app; then
```

Checks for both application containers running.